control/HTTP.php

```diff
@@ -318,139 +319,117 @@ public static function register_etag($etag) {
 	 *                            deprecated; in these cases, the headers are output directly.
 	 */
 	public static function add_cache_headers($body = null) {
-		$cacheAge = self::$cache_age;
 
 		// Validate argument
 		if($body && !($body instanceof SS_HTTPResponse)) {
 			user_error("HTTP::add_cache_headers() must be passed an SS_HTTPResponse object", E_USER_WARNING);
 			$body = null;
 		}
 
-		// Development sites have frequently changing templates; this can get stuffed up by the code
-		// below.
-		if(Director::isDev()) $cacheAge = 0;
```

Comment:
I think given the security implications of getting this change wrong, it'd be good to have quite granular commits for specific stuff, for example one commit for "move isDev below headers_sent check" to make that explicit.

Reply:
I've split this PR into smaller commits that are hopefully a bit easier to follow